File: MultiLanguageSandbox/src/test/deps/swift/42.swift
func asciiCode(of character: Character) -> Int {
    return Int(character.unicodeScalars.first!.value)
}


func testAsciiCode() {
    assert(asciiCode(of: "A") == 65, "Test Case 1 Failed")
    assert(asciiCode(of: "a") == 97, "Test Case 2 Failed")
    assert(asciiCode(of: "0") == 48, "Test Case 3 Failed")
    assert(asciiCode(of: "#") == 35, "Test Case 4 Failed")
    assert(asciiCode(of: "Z") == 90, "Test Case 5 Failed")
    assert(asciiCode(of: "z") == 122, "Test Case 6 Failed")
}

testAsciiCode()

File: MultiLanguageSandbox/src/test/deps/elixir/42.exs
defmodule AsciiCode do
def get_ascii_code(char) when is_binary(char) and byte_size(char) == 1 do
    # Check if the character is a space (ASCII 32)
    if char == " " do
      raise ArgumentError, "Character cannot be a space"
    else
      # Get the ASCII code by converting the binary to a list of integers
      [code] = :binary.bin_to_list(char)
      code
    end
  end
end



ExUnit.start()
defmodule AsciiCodeTest do
  use ExUnit.Case

  test "get_ascii_code returns correct ASCII code for uppercase letters" do
    assert AsciiCode.get_ascii_code("A") == 65
    assert AsciiCode.get_ascii_code("Z") == 90
  end

  test "get_ascii_code returns correct ASCII code for lowercase letters" do
    assert AsciiCode.get_ascii_code("a") == 97
    assert AsciiCode.get_ascii_code("z") == 122
  end

  test "get_ascii_code returns correct ASCII code for digits" do
    assert AsciiCode.get_ascii_code("0") == 48
    assert AsciiCode.get_ascii_code("9") == 57
  end

  test "get_ascii_code returns correct ASCII code for special characters" do
    assert AsciiCode.get_ascii_code("!") == 33
    assert AsciiCode.get_ascii_code("@") == 64
  end
end